src/PortfolioPlanning/Redux/Reducers/EpicTimelineReducer.ts

```diff
@@ -185,6 +197,24 @@ function handlePortfolioItemsReceived(
                         countProgress: newItemInfo.CountProgress
                     });
                 }
+
+                if (noDatesFound) {
+                    const doc: JsonPatchDocument = [
+                        {
+                            op: "add",
+                            path: "/fields/Microsoft.VSTS.Scheduling.StartDate",
+                            value: now
+                        },
+                        {
+                            op: "add",
+                            path: "/fields/Microsoft.VSTS.Scheduling.TargetDate",
+                            value: oneMonthFromNow
+                        }
+                    ];
+
+                    const witHttpClient: WorkItemTrackingHttpClient = VSS_Service.getClient(WorkItemTrackingHttpClient);

```

Comment:
I didn't notice this until now, but this function is getting called by the reducer. Which means the reducer has side effects... That breaks the rules of redux. 

We need to move all of these side effects that call the client or that do things with the storage to a saga or a separate function.

Reply:
Looks like the rest of the reducer is fine, it's just your addition of using the client.

Reply:
Maybe we can set the default dates in the function that fetches the epics in the first place?

Reply:
One place I suggest could be the `EpicTimelineSaga.onAddEpics` ... here we can check if the epics to add contain dates, and if not save default ones using the WIT client before updating the extension storage with the work item ids.